elastic/kibana
# [ML] Add tests to create saved search based jobs

## Summary

This PR adds tests for ML job creation based on saved searches (5 scenarios: filter, lucene query, kuery query, filter + lucene query, filter + kuery query).

### High level test steps

- Create multi metric job based on saved search
- Verify detector preview takes the saved search into account (number of split cards, split card title)
- Verify job details after creation, especially the processed records to see if the saved search has been applied correctly

### Other changes

- Add saved searches to ml/farequote archive and clean up some not needed fields
- Create a new test folder `anomaly_detection` and move existing job wizard tests into it
- Update job source selection service after reactification
